Sources/protoc-gen-swift/GeneratorOptions.swift

```diff
@@ -89,6 +114,8 @@ class GeneratorOptions {
       visibilitySourceSnippet = ""
     case .public:
       visibilitySourceSnippet = "public "
+    case .spi(let identifier):
+      visibilitySourceSnippet = "@_spi(\(identifier)) public "
```

Comment:
Will this start spraying `@_spi` everywhere? Would we want it to appear only on type declarations and extension members?

Reply:
It will spray everywhere and we also need it to. It is kinda like inlinability once you start adding it, everything needs to become spi.

Reply:
Do we have an opinion about what happens when the string is not a swift identifier?

Reply:
Users could supply bad inputs here like (); however, I don't think it is worth validating anything on our side. First, this should only be used by people who know what they want. Second, it will result in compiler errors when building the generated code.

Reply:
You can use [`isValidSwiftIdentifier`](https://github.com/apple/swift-protobuf/blob/a7c57dc39883e15b2f0f513e2f36ff117e131b58/Sources/SwiftProtobufPluginLibrary/SwiftLanguage.swift#L96) to do some very basic validation here if you want.

Reply:
I think it is worth validating whether this is an identifier before we write it. Earlier errors are better.